cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java

```diff
@@ -330,6 +328,41 @@ private ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline validat
     return specBuilder;
   }
 
+  private Object getPlugin(String stageName, ETLPlugin etlPlugin, PluginSelector pluginSelector, String type,
+                           String pluginName, DefaultStageConfigurer stageConfigurer) {
+    Object plugin = null;
+    try {
+      plugin = pluginConfigurer.usePlugin(type, pluginName, stageName, etlPlugin.getPluginProperties(), pluginSelector);
+    } catch (IllegalArgumentException e) {
+      stageConfigurer.addFailure(e.getMessage(), String.format("Provide unique plugin id for plugin '%s'.", stageName));
```

Comment:
Had an offline discussion with @albertshau . It makes sense to not catch pipeline app generated exceptions and let it propagate.

Reply:
The implementations of `pluginConfigurer.usePlugin` method does not handle exceptions correctly.  Filed: https://issues.cask.co/browse/CDAP-15744 to fix it. 
For now, adding validation failure by capturing `Exception` instead of specific exceptions.